R/grid.R

```diff
@@ -0,0 +1,295 @@
+#' Generate CSS string representing grid template areas.
+#' @param areas_dataframe data.frame of strings representing grid areas
+#' @return string
+#' @details This is a helper function used in grid_template()
+#' \preformatted{
+#' areas_dataframe <- rbind(
+#'    c("header", "header", "header"),
+#'    c("menu",   "main",   "right1"),
+#'    c("menu",   "main",   "right2")
+#' )
+#'
+#' result == "'header header header' 'menu main right1' 'menu main right2'"
+#' }
+#'
+data_frame_to_css_grid_template_areas <- function(areas_dataframe) {
+  apply(areas_dataframe, 1, function(row) paste(row, collapse = ' ')) %>%
+    lapply(., function(row) glue::glue("'{row}'")) %>%
+    paste(., collapse = ' ')
+}
+
+#' Generate template string representing CSS styles of grid container div.
+#'
+#' @param css_grid_template_areas string, CSS value for grid-template-areas
+#' @param rows_height vector of strings
+#' @param cols_width vector of strings
+#'
+#' @return string
+#' @details This is a helper function used in grid_template()
+#' \preformatted{
+#'   grid_container_css(
+#'     "'a a a' 'b b b'",
+#'     c("50\%", "50\%"),
+#'     c("100px", "2fr", "1fr")
+#'   )
+#' }
+#' returns
+#' \preformatted{
+#'   "display: grid;
+#'    height: 100\%;
+#'    grid-template-rows: 50\% 50\%;
+#'    grid-template-columns: 100px 2fr 1fr;
+#'    grid-template-areas: 'a a a' 'b b b';
+#'    {custom_style_grid_container}"
+#' }
+#'
+grid_container_css <- function(css_grid_template_areas, rows_height, cols_width) {
+  grid_container_styles <- c(
+    "display: grid",
+    "height: 100%",
+    glue::glue("grid-template-rows: {paste(rows_height, collapse = ' ')}"),
+    glue::glue("grid-template-columns: {paste(cols_width, collapse = ' ')}"),
+    glue::glue("grid-template-areas: {css_grid_template_areas}"),
+    "{custom_style_grid_container}" # To be rendered by glue::glue() later
+  )
+  paste(grid_container_styles, collapse = "; ")
+}
+
+#' Generate list of HTML div elements representing grid areas.
+#'
+#' @param area_names
+#'
+#' @return list of shiny::tags$div
```

Comment:
could you wrap `shiny::tags$div` in `\code`?

Reply:
done :heavy_check_mark: